### PR TITLE
 Modifying parallel search algorithm for zero length searches

### PR DESCRIPTION
**Pull request by dcbdan**
Changed search.hpp to return the first iterator if the search iterators are the same.

### CODE REVIEW

File: hpx/parallel/algorithms/search.hpp [hunk 1]
```diff
@@ -66,7 +66,7 @@ namespace hpx {namespace parallel { HPX_INLINE_NAMESPACE(v1)
 
                 s_difference_type diff = std::distance(s_first, s_last);
                 if (diff <= 0)
-                    return result::get(std::move(last));
+		  return result::get(std::move(first));
```
Comment (hkaiser) [inline hpx/parallel/algorithms/search.hpp]: I think this messed up the indentation. Please make sure that your editor is not inserting tabs into the files (see: https://github.com/STEllAR-GROUP/hpx/wiki/HPX-Source-Code-Structure for general coding guidelines).

**Comment by dcbdan** [comment]
Fixed my editor to use .editorconfig file. Indentation should be correct.

**Comment by hkaiser** [comment]
Looks good to me.

**Comment by hkaiser** [comment]
Thanks!

**Pull request merged**